### PR TITLE
GHA: add heffte builds

### PR DESCRIPTION
**Pull request by junghans**
Part of #345

### CODE REVIEW

**Comment by junghans** [comment]
Needs https://github.com/kokkos/ci-containers/pull/8

File: .github/workflows/CI2.yml [hunk 1]
```diff
@@ -21,107 +21,130 @@ jobs:
...
         coverage: ['OFF']
         include:
           - distro: 'fedora:intel'
             cxx: 'icpc'
             openmp: 'ON'
             cmake_build_type: 'Release'
             kokkos_ver: 'master'
-            arborx: ['NoArborX']
+            arborx: 'NoArborX'
+            heffte: 'NoheFFTe'
```
Comment (streeve) [inline .github/workflows/CI2.yml:33]: We should add to at least one intel build with MKL

**Comment by junghans** [comment]
@streeve the containers works now, can you try your luck on the mkl builds?

File: .github/workflows/CI2.yml [hunk 2]
```diff
@@ -166,14 +205,33 @@ jobs:
...
+        run: |
+          git clone --depth 1 --branch v2.0.0 https://bitbucket.org/icl/heffte.git heffte
+      - name: Build heffte
+        if: ${{ matrix.heffte != 'NoheFFTe' }}
+        working-directory: heffte
+        run: |
+          cmake -B build \
+            -DCMAKE_CXX_STANDARD="11" \
+            -DBUILD_SHARED_LIBS=ON \
+            -DHeffte_ENABLE_${{ matrix.heffte }}=ON \
```
Comment (junghans) [inline .github/workflows/CI2.yml:220]: Uh, that is ugly.

Comment (streeve) [reply .github/workflows/CI2.yml:220]: I don't love it. Would you prefer to build the list of optional flags like we used to with travis?

Comment (junghans) [reply .github/workflows/CI2.yml:220]: we could do the same in shell above
```
[[ ${{ matrix.heffte }} = "heFFTe_with_FFTW" ]] && cmake_opts+=(  -DHeffte_ENABLE_FFTW=ON )
[[ ${{ matrix.heffte }} = "heFFTe_with_MKL" ]] && cmake_opts+=(  -DHeffte_ENABLE_MKL=ON )
```

**Review by junghans — COMMENTED** [review]
Looks good now!

**Comment by junghans** [comment]
Intel fails with:
```
43 - Cabana_NeighborList_test_SERIAL (Failed)
```

**Comment by junghans** [comment]
@streeve any idea?

**Comment by streeve** [comment]
Here's what Miro said about the tests with heFFTe MKL build failures:
> The problem is with the newer versions of the Intel compiler and the way that they infer and convert template parameters. I will have to add a work-around and be explicit, which is not the most convenient thing. Clang, Gcc, and XL work fine, so I have to work-around an outlier compiler (frustrating).
If you can downgrade the version in the Intel compiler that will give you a temporary work-around.

We can discuss what we want to do with those tomorrow

**Comment by streeve** [comment]
@sslattery ready for review, then we can reset the required tests again

**Review by sslattery — APPROVED** [review]

**Pull request merged**